### PR TITLE
Multithreaded CI builds

### PR DESCRIPTION
**Pull request by netrunnereve**
The CUDA install is still super slow, but if you have the threads you might as well use them! This halves build time for some targets.

### CODE REVIEW

**Comment by Cebtenzzre** [comment]
Is there any reason we shouldn't use `-j $(nproc)` on Linux and `-j $(sysctl -n hw.logicalcpu)` on macOS?

**Comment by staviq** [comment]
> Is there any reason we shouldn't use `-j $(nproc)` on Linux and `-j $(sysctl -n hw.logicalcpu)` on macOS?

I second this, since adding self hosted runners on forks for testing PRs is trivial, detecting local CPU count can be very useful.

While we're at it,

GitHub appears to automatically prioritize self hosted runners, and the only thing you need to do is to give them the same tag `windows-latest` etc. Docker isn't needed, self hosted runners work just fine on persistent image ( normal computer with normal OS ), and that makes subsequent jobs even faster too.

As soon as you register a self hosted runner in your forked repo, CI jobs will run on it automatically.

**Comment by netrunnereve** [comment]
> Is there any reason we shouldn't use `-j $(nproc)` on Linux and `-j $(sysctl -n hw.logicalcpu)` on macOS?

Done, and added for Windows too. That's a much better way of doing it.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -472,7 +472,7 @@ jobs:
         run: |
             sudo pkg update
             sudo pkg install -y gmake automake autoconf pkgconf llvm15 clinfo clover opencl clblast openblas
-            gmake CC=/usr/local/bin/clang15 CXX=/usr/local/bin/clang++15
+            gmake CC=/usr/local/bin/clang15 CXX=/usr/local/bin/clang++15 -j `sysctl -n hw.ncpu`
```
Comment (Green-Sky) [inline .github/workflows/build.yml:475]: any reason you switched syntax for this one?

Comment (netrunnereve) [reply .github/workflows/build.yml:475]: The FreeBSD shell doesn't support the `$()` syntax.

**Review by ErikBjare — APPROVED** [review]
LGTM!

File: .github/workflows/build.yml (hunk 1 above)
Comment (Green-Sky) [reply .github/workflows/build.yml:475]: I see. Should we switch all the others then?
This is probably the difference between `sh` and `bash`

Comment (Cebtenzzre) [reply .github/workflows/build.yml:475]: > This is probably the difference between `sh` and `bash`

Nope. Modern POSIX sh supports `$()`, you can try it out for yourself with the dash shell. FreeBSD defaults to tcsh which doesn't support `$()`.

**Review by Green-Sky — APPROVED** [review]

**Review by ggerganov — APPROVED** [review]

**Pull request merged**